### DISCUSSION TITLE
Load other page without unloading unity player

### DISCUSSION THREAD

**Discussion by jayden-tan-sg** [jeffreylanters/react-unity-webgl#261] [Question and Answer]
Is there a way to navigate or overlay another page without unloading unity webgl?

My webgl load time is around 30 sec even after caching, wanted to improve user experience as user will frequently navigate to other pages and back to unity 

my webgl project is 3d serious game application whereby needs to show multiple react dashboard.

**Comment by jeffreylanters** [comment]
Hi. Thanks for your question. 

No there is no way to disable this behavior. You could simply avoid the player being unloaded by, for example placing it outside of your router so it is always being rendered and then changing is visibility using CSS.

**Comment by Edvinas01** [comment]
I'm facing a similar issue. In my case I cannot move the `Unity` component up the tree as I have to change the layout of the app in each route quite significantly.

To illustrate this better, I have some routes which render a side-bar on the left side of the app while some routes render side-bars on the right:
```tsx
// Component A
render() {
  return (
    <>
      <LeftSideBar ... />
      <Unity ... />
    </>
  )
}

// Component B
render() {
  return (
    <>
      <LeftSideBar ... />
      <Unity ... />
      <RightSideBar ... />
    </>
  )
}

// Component C
render() {
  return (
    <>
      <Unity ... />
      <RightSideBar ... />
    </>
  )
}
```

Additionally, I have more routes which need to render things on top of the `Unity` component which complicates matters even more.

Are there any ways I could workaround this by preventing the player from unloading? I could fork the repo and make some changes for my use-case, though unsure if this is possible at all.
